kraken/network_scenarios/actions.py

```diff
@@ -0,0 +1,83 @@
+import yaml
+import logging
+import time
+import kraken.cerberus.setup as cerberus
+import kraken.kubernetes.client as kubecli
+import kraken.node_actions.common_node_functions as common_node_functions
+
+
+pod_yml = """
+apiVersion: v1
+kind: Pod
+metadata:
+  name: fedtools
+spec:
+  hostNetwork: true
+  nodeName: {nodename}
+  containers:
+  - name: fedtools
+    image: docker.io/fedora/tools
+    command:
+    - /bin/sh
+    - -c
+    - |
+      sleep infinity
+    securityContext:
+      privileged: true
+    volumeMounts:
+      - mountPath: /lib/modules
+        name: lib-modules
+        readOnly: true
+  volumes:
+    - name: lib-modules
+      hostPath:
+        path: /lib/modules
+"""
+
+
+# Reads the scenario config and introduces traffic variations in Node's host network interface.
+def run(scenarios_list, config, wait_duration):
+    failed_post_scenarios = ""
+    logging.info("Runing the network scenarios tests")
+    for net_config in scenarios_list:
+        with open(net_config, "r") as file:
+            param_lst = ["latency", "loss", "bandwidth"]
+            param_map = {"latency": "delay", "loss": "loss", "bandwidth": "rate"}
+            test_config = yaml.safe_load(file)
+            test_dict = test_config["network_scenarios"]
+            test_duration = int(test_dict.get("duration", 300))
+            test_interface = test_dict["interfaces"]
```

Comment:
Can you set a default value here, think [] is good

Reply:
Since "interface" is required field, rather then relying on Exception, will rewrite the code to evaluate the variable and exit if it is not set.